### PR TITLE
Add challenge 58

### PR DESCRIPTION
**Pull request by svkirillov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+Terms of challenge: [challenge58.txt](./tasks/challenge58.txt)
+
+Run test for Pollard's Method for Catching Kangaroos Algorithm:
+
+```sh
+go test -v -count=1 ./challenge58 -run TestCatchWildKangaroo
+```
+
+
+Run quick test for Catching Kangaroos Attack:
```
Comment (dnkolegov) [inline README.md:14]: a quick test

File: README.md [hunk 2]
```diff
@@ -1 +1,24 @@
 # cryptopals-go
+
+## Challenge 58
+
+Terms of challenge: [challenge58.txt](./tasks/challenge58.txt)
+
+Run test for Pollard's Method for Catching Kangaroos Algorithm:
```
Comment (dnkolegov) [inline README.md:7]: a test

File: README.md [hunk 3]
```diff
@@ -1 +1,24 @@
...
+```
+
+
+Run quick test for Catching Kangaroos Attack:
+
+```sh
+go test -v -count=1 ./challenge58 -run TestCatchingKangaroosAttackFast
+```
+
+Run slow test for Catching Kangaroos Attack:
```
Comment (dnkolegov) [inline README.md:20]: a slow

File: README.md (hunk 1 above)
Comment (svkirillov) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (svkirillov) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (svkirillov) [reply README.md]: fixed

**Comment by svkirillov** [comment]
Complete

**Pull request merged**